pandas/compat/__init__.py

```diff
@@ -74,11 +74,9 @@ def raise_with_traceback(exc, traceback=Ellipsis):
     raise exc.with_traceback(traceback)
 
 
-# In Python 3.7, the private re._pattern_type is removed.
-# Python 3.5+ have typing.re.Pattern
 if PY36:
-    import typing
-    re_type = typing.re.Pattern
+    from typing import Pattern, Type
```

Comment:
do these exist in 3.5? can u move this import to the top?

Reply:
Type I think is from 3.5.2. We've used elsewhere so need to address at some point. Maybe vendoring, either dropping < 3.5.2 or vendoring that piece

Reply:
Ignoring my previous comment for a second would we even need the Py36 condition or can we import all from the same space across versions?

Reply:
I've successfully tested `from typing import Pattern, Type` in 3.5.7, 3.6.8, and 3.7.3. I can't speak for anything before 3.5.7, but there's no problem importing after the most recent 3.5 micro version. I'll move the imports to the top and drop the else.